src/server/common.h

```diff
@@ -268,7 +268,10 @@ class Context : protected Cancellation {
   //
   // Note: this function blocks when called from inside an error handler.
   template <typename... T> GenericError Error(T... ts) {
-    std::lock_guard lk{mu_};
+    if (!mu_.try_lock())  // TODO: Maybe use two separate locks.
```

Comment:
can you explain why try lock?

Reply:
Because it happened so that `Error()` was called from inside the error handler. We should maybe allow this in the future, this is why I've added a TODO